en/1/02-contracts.md

```diff
@@ -0,0 +1,77 @@
+---
+title: "Contracts"
+actions: ['checkAnswer', 'hints']
+material: 
+  editor:
+    language: rust
+    startingCode: |
+      #![no_std]
+
+      multiversx_sc::imports!();
+      multiversx_sc::derive_imports!();
+    answer: > 
+      #![no_std]
+
+      multiversx_sc::imports!();
+      multiversx_sc::derive_imports!();
+
+      #[multiversx_sc::contract]
+      pub trait ZombiesContract {
+        #[init]
+        fn init(&self) {}
+      }
+---
+
+
+Starting with the absolute basics:
+
+Rust is a multiparadigm  programming language focused on type safety and performance, enabling users to build fast and robust applications. A `contract` is the fundamental building block of MultiversX applications — all variables and functions belong to a contract, and this will be the starting point of all your projects.
+
+An empty contract named `HelloWorld` can be created with **mxpy** by running `mxpy contract new hello-world --template empty` from terminal and its structure would look like this:
+
+```
+├── meta
+├── scenarios
+├── src
+  ├── lib.rs
+├── tests
+├── wasm
+. Cargo.toml
+```
+In here we are interested for now of **lib.rs** from inside the root folder **src**, and would look like this.
+
+```
+#[mx_sc::contract]
+pub trait ZombiesContract {
+        #[init]
+        fn init(&self) {}
+}
+```
+
+Here `#[multiversx_sc::contract]` is a procedural macro that defines the `ZombieContract` trait as a contract. You will see a pattern of using procedural macros, since these are the ones that make the transition from a more user friendly code writing to a very complex and high performant contract build.
```

Comment:
How about something like this: `The first line of code" `#[multiversx_sc::contract]` is a procedural macro that defines the `ZombieContract` trait as a contract. You will see a pattern of using procedural macros, since these are the ones that take care of optimizing the code you write. Think of them for now as some additional annotations that transform basic rust elements into smart contract ones.`